src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -3620,6 +3620,54 @@ public void TestAttachmentDottedNameNodeBinding(string script, string expectedSc
             symbol.AddEntity(new TestDataSource("DS", schema));
 
             TestSimpleBindingSuccess(script, expectedType, symbol);
+        }
+
+        [Theory]
+        [InlineData("Search(T, \"a\", \"Name\")", "*[Name:s]")]
+        [InlineData("Search(T, If(1<0,\"a\"), \"Name\")", "*[Name:s]")]
+        [InlineData("Search(T2,\"a\", \"Name\", \"Address\")", "*[Name:s,Age:n,Address:s]")]
+        public void TexlFunctionTypeSemanticsSearch(string script, string expectedType)
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[Name:s]")));
+            symbol.AddVariable("T2", new TableType(TestUtils.DT("*[Name:s,Age:n,Address:s]")));
+
+            TestSimpleBindingSuccess(
+                script,
+                TestUtils.DT(expectedType),
+                symbol,
+                new Features { ConsistentOneColumnTableResult = true });
+        }
+
+        [Theory]
+
+        // missing 3rd arg
+        [InlineData("Search(T, \"a\")", "*[]")]
+
+        // Age is number, only string supported.
+        [InlineData("Search(T2, \"a\", \"Age\"", "*[Address:s, Age:n, Name:s]")]
+        [InlineData("Search(T3, \"a\", \"Age\"", "*[Age:n]")]
+
+        // 3rd onwards Arg should always be constant string.
+        [InlineData("Search(T2, \"a\", If(1<0, \"Name\")", "*[Address:s, Age:n, Name:s]")]
+
+        // 2nd Arg should only be a string.

```

Comment:
>// 2nd Arg should only be a string. [](http://example.com/codeflow?start=8&length=35)

Should we allow coercion in the second argument? In all functions that expect a string value we let other primitive types to be passed.

Reply:
We could!
I am not worried abt semantic change here as its widening the restriction.
Just concerned as to end users may end up getting tricked in it if they don't fully grasp the coercion!
e.g. `Search(table, Date(), column,..)` would start working and they would need to know what exactly Text(Date()) would actually look like...
cc @MikeStall any thoughts?